exec/java-exec/src/main/java/org/apache/drill/exec/planner/physical/ConvertCountToDirectScan.java

```diff
@@ -103,6 +104,10 @@ public void onMatch(RelOptRuleCall call) {
       return;
     }
 
+    if (proj != null && !ProjectRemoveRule.isTrivial(proj)) {
```

Comment:
I feel that this check might over-kill some optimization opportunity.  For example,

select count(100) 
from `parquetTable`;

In this case, count(100) is equal to rowcount in parquet table. However, the project is not a trial project, meaning the new code will disable the optimization.